src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -130,6 +130,18 @@ public void AddProject(PackageSpec projectSpec)
             _projects.Add(projectUniqueName, projectSpec);
         }
 
+        /// <summary>
+        /// Add all restorable projects to the restore list.
+        /// This is the behavior for --recursive
+        /// </summary>
+        public void AddRestoreForAllProjects()
+        {
+            // Add everything from projects except for packages.config and unknown project types
+            _restore.UnionWith(Projects.Where(project =>
```

Comment:
For instance caller method avoids adding _auxiliary_ projects via this method when no original restore project(s) exists. So I was wondering if this pre-condition should be added here as well.